drivers/i2c/i2c_ll_stm32.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016 BayLibre, SAS
+ * Copyright (c) 2017 Linaro Ltd
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#ifndef _STM32_I2C_H_
+#define _STM32_I2C_H_
+
+typedef void (*irq_config_func_t)(struct device *port);
+
+struct i2c_stm32_config {
+#ifdef CONFIG_I2C_STM32_INTERRUPT
+	irq_config_func_t irq_config_func;
+#endif
+	struct stm32_pclken pclken;
+	I2C_TypeDef *i2c;
+};
+
+struct i2c_stm32_data {
+#ifdef CONFIG_I2C_STM32_INTERRUPT
+	struct k_sem device_sync_sem;
+#endif
+	union dev_config dev_config;
+#ifdef CONFIG_I2C_STM32_V1
+	u16_t slave_address;
+	struct {
+		unsigned int is_restart;
+		unsigned int flags;
+#else
+	struct {
+#endif
+		unsigned int is_write;
+		unsigned int is_nack;
+		unsigned int is_err;
+		struct i2c_msg *msg;
+		unsigned int len;
+		u8_t *buf;
+	} current;
+};
+
+s32_t msg_write(struct device *dev, struct i2c_msg *msg, u32_t flg, u16_t sadr);
+s32_t msg_read(struct device *dev, struct i2c_msg *msg, u32_t flg, u16_t sadr);
+s32_t i2c_configure_timing(struct device *dev, u32_t clk);
+void i2c_stm32_event_isr(void *arg);
+void i2c_stm32_error_isr(void *arg);
+
+#define DEV_DATA(dev) ((struct i2c_stm32_data * const)(dev)->driver_data)
+#define NEXT(p, len) do { p++; len--; } while (0)
```

Comment:
lets remove the NEXT macro, I think it makes it hard to understand what's going on inside the code.